### PR TITLE
LAB-1192: exposure controls — default-deny auth, operator-only /_stats + /metrics, real client IP behind an LB, failed-auth throttling

### PR DESCRIPTION
**Pull request by 27Bslash6**
## Summary

This PR implements exposure controls (LAB-1192) to harden the proxy's security posture, converting the previously default-open authentication model to default-deny. It adds four distinct mechanisms.

### 1. Default-deny authentication
- The proxy now **fails at startup** if no credentials are configured (neither `[[clients]]` nor `proxy_key`).
- Introduces a single escape hatch, `allow_unauthenticated = true`, for trusted-network-only deployments (e.g., behind a Kubernetes NetworkPolicy or tailnet). This flag is mutually exclusive with configured credentials.
- Adds a minimum credential-length floor (`MIN_KEY_LEN = 32`), rejecting any `proxy_key` or client key shorter than 32 characters at startup with a message suggesting `openssl rand -hex 32`.
- New `validate_exposure()` function performs these startup checks.

### 2. Operator-only admin surfaces
- `/_stats` and `/metrics` are now gated to **operator** principals only under `[[clients]]`: unauthenticated requests get `401`, valid non-operator credentials get `403`.
- Under legacy `proxy_key`, the shared key holder is treated as the operator; under `allow_unauthenticated`, both surfaces stay open with a rate-limited `warn` log per access.
- New `authorize_admin()` helper enforces this. A startup warning fires when `[[clients]]` is configured with an empty `operators` list, since that would block all monitoring scrapes.

### 3. Real client IP resolution behind a load balancer
- Adds `trusted_proxies` config for load balancers whose `x-forwarded-for` header is honored.
- New `resolve_client_ip()` function: when the TCP peer is a trusted proxy, the client IP becomes the rightmost XFF entry not itself in the trusted list; otherwise the peer address is used and the header is ignored entirely. Malformed entries fall back to the peer address.
- Canonicalizes IPv4-mapped IPv6 peers before the trust check to avoid silently disabling XFF resolution on dual-stack listeners.
- All handlers now call this function exactly once and thread the resolved IP through the allowlist, auth, budgets, throttle, and logs. A startup warning fires when credentials exist but no `trusted_proxies` is set.

### 4. Failed-auth throttling
- Adds `auth_failure_limit` (default 10) and `auth_failure_window_secs` (default 300) config.
- New `AuthThrottle` struct: after the limit of failed attempts from a client IP within the window, further requests get `429` with `retry-after` **before** any key comparison (closing the timing side-channel).
- State is bounded at 4096 IPs with an eviction policy that purges expired windows first and preserves established lockouts, preventing the tracking structure itself from being used as a DoS vector.
- Successful authentication clears an IP's failure state to avoid shared-IP false lockouts.
- Failures are counted in a new Prometheus metric `anthropic_auth_failures_total{route}`.

### Documentation & tests
- Updated `CLAUDE.md`, `README.md`, and `config.toml.example` to document all new config options, the default-deny posture, TLS-termination guidance, and updated admin-surface behavior; removed the now-resolved "not operator-gated" known limitation.
- Added ~30 tests covering startup posture validation, XFF resolution edge cases, the operator admin-access matrix, and throttle behavior including capacity/eviction and reset-on-success.

## Summary by CodeRabbit

* **New Features**
  * Added trusted-proxy support for accurate client IP detection.
  * Added configurable throttling for repeated authentication failures.
  * Added route-specific authentication metrics.
  * Restricted administrative observability endpoints to operator credentials.
  * Added default-deny authentication with explicit controls for unauthenticated access.
  * Added startup validation for authentication and security settings.

* **Documentation**
  * Expanded configuration and security guidance covering proxy handling, credential requirements, IP allowlists, authentication throttling, and monitoring access.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -307,18 +328,38 @@ It **fails closed** on a model it cannot read. The proxy takes the model from th
...
+
+**TLS terminates at the ingress.** The proxy speaks plain HTTP and its container port must never be published directly to the internet — put it behind a TLS-terminating load balancer or ingress, list that LB in `trusted_proxies`, and let the ingress carry the certificate. Bearer credentials without TLS are credentials in cleartext.
+
+### Admin surfaces are operator-only (LAB-1192)
+
+`/_stats` disclosures are a reconnaissance report for anyone planning to spend the pool: raw `client_id`s, agent/session prefixes, models, **endpoint account names**, token counts, per-account utilisation and budgets. So under `[[clients]]` both `/_stats` and `/metrics` answer only to a client named in `operators` — unauthenticated gets `401`, a valid non-operator credential gets `403`. Under legacy `proxy_key` the (single) key holder is the operator by construction. Under `allow_unauthenticated` both surfaces serve, and every access logs at `warn` so the open posture stays visible.
```
[the remark this answers is not in the document]
Comment (27Bslash6) [reply README.md]: Fixed in cbcf9a1 — all three files now state the warn is rate-limited to once per route per 5 minutes (OPEN_ADMIN_WARN_INTERVAL).

File: README.md [hunk 2]
```diff
@@ -307,18 +328,38 @@ It **fails closed** on a model it cannot read. The proxy takes the model from th
...
+
+> [!IMPORTANT]
+> Behind a load balancer, **`[[clients]]` credentials are the identity**. The `client_names` IP map is a lab-only convenience: it maps *source addresses*, and once traffic arrives through an LB the recovered XFF address is only as trustworthy as the LB's own header hygiene. Do not hang budgets or operator status on `client_names` on a public ingress.
+
+### Failed-auth throttling (LAB-1192)
+
+A static bearer credential on the public internet gets scanned. After `auth_failure_limit` failures from one client IP inside `auth_failure_window_secs`, further requests from that IP get `429` with `retry-after` **before any key comparison runs** — a locked-out guesser gets nothing back, not even timing, until the window expires. Failures are counted in `anthropic_auth_failures_total{route}` and logged with the resolved client IP. The throttle table is bounded (4096 IPs, oldest-window eviction), so the tracking structure itself cannot be flooded into an OOM.
```
[the remark this answers is not in the document]
Comment (27Bslash6) [reply README.md]: Fixed in cbcf9a1 — README and CLAUDE.md now describe the actual policy: expired-window purge first, then least-established eviction (lowest failure count, oldest window as tie-breaker).

**Comment by 27Bslash6** [comment]
@coderabbitai review
